### PR TITLE
Ignore nightly tag in nightly workflow as well

### PR DESCRIPTION
**Pull request by kmilos**
Complements #14665

### CODE REVIEW

**Comment by victoryforce** [comment]
@kmilos It would be better not to fix such code in all places, but to replace it with a call to get_git_version_string.sh. Without such deduplication, the next time one can again forget about the need for corrections in many places.

**Comment by kmilos** [comment]
Sure, good idea. Unfortunately I won't have time to do it and extensively test it in the coming days...

**Comment by victoryforce** [comment]
> Sure, good idea. Unfortunately I won't have time to do it and extensively test it in the coming days...

OK. Your fix is so simple it's obviously correct. @TurboGit I suggest to accept it ASAP, I will do the deduplication of the code later. Also, after accepting this PR but before running tonight's nightly, I'd also like to propose a PR that will keep the nightly tag in the repository. Now that nightly will not interfere with building the version string correctly, it would be better not to remove it. I will explain why in the next comment.

**Comment by victoryforce** [comment]
@TurboGit Now that we've fixed the versioning issues, I'm pretty sure we shouldn't be removing the nightly tag from the repository. This removal was the first reaction to the problem, but with recent fixes in the scripts we get the correct version even WITH the nightly tag in the repository. Yes, the test showed that even if the tag is removed, the artifacts will remain on the Releases page. However, there are several disadvantages:
- The fact that the artifacts of a release with a removed tag are not completely removed from the Releases page is a feature of the current Github implementation and this may change! I don't think we should rely on it.
- The possibility of emoji reactions under the release is lost. Not that it really matters, but...
- The possibility to have a permanent URL specifically for nightly builds is lost. Not that it's critically necessary, but it would be nice.
- Next to the release box there are useful links: the ability to compare the build tag with other tags, a reference to the last commit, a link to the state of the repository with the build tag. When the nightly tag is removed, they are lost, but it would be good to have them!

_Many_ repositories on Github have nightly tags. Yes, a normal git pull doesn't update these tags (users have to update it with a separate command if it is important for their clone for some reason), but for these projects, it's not something worth paying attention to or fixing.

In summary, yes we can (for the time being) have a release with the tag removed, but it doesn't seem like a good and reliable solution.

**Comment by TurboGit** [comment]
@victoryforce : Please open a separate issue for this discussion to not be lost as I'll merge this now. TIA.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by TurboGit** [comment]
@victoryforce : OK I see that you already have a PR referencing your comment. No need for an issue at this point.
